src/rez/cli/test.py

```diff
@@ -60,10 +77,19 @@ def command(opts, parser, extra_arg_groups=None):
     if opts.TEST:
         run_test_names = opts.TEST
     else:
-        run_test_names = test_names
+        # if no tests are explicitly specified, then run only those with a
+        # 'default' run_on tag
+        run_test_names = runner.get_test_names(run_on=["default"])
+
+        if not run_test_names:
+            print(
+                "No tests with 'default' run_on tag found in %s" % uri,
+                file=sys.stderr
+            )
+            sys.exit(0)
 
     for test_name in run_test_names:
-        returncode = runner.run_test(test_name)
+        if not runner.stopped_on_fail:
+            runner.run_test(test_name)
 
-        if returncode:
-            sys.exit(returncode)
```

Comment:
It's intentional, as it isn't clear what this would mean anymore.

Previously, rez-test raised as soon as a test fails. It doesn't do that anymore (unless you specify --stop-on-fail), and it also iterates over variants. So now, you get a summary of all the tests, showing whether they passed/failed/skipped. In that context it isn't really clear what the exit code should be, or if it's useful to have at all.

Reply:
I have the feeling it becomes unclear because the tests are skipped instead of being marked as failed.  I've been thinking about it even more today and I really think tests that fail to resolve should really be marked as failed. That would avoid hiding possible issues and would probably make it possible to use a different exit code more easily.

A lot of CI system base themselves on the exit code of processes to know if a step failed or not. If we don't return a proper exit code, it's going to be quite difficult to know when a test failed or not and mark a build as failed.

Reply:
Ok I'll mark those as failed, and set exit code to match exit code of first failed test.